File: p1/TreeNode.c
// Tree node holds the actual integers
// and containers pointers to its parent,
// left child, and right child

#include <stdio.h>
#include <stdlib.h>
#include "TreeNode.h"

TreeNode *newTreeNode(int value, TreeNode *parent) {
    TreeNode *tn = malloc(sizeof(TreeNode));
    if (tn == 0) { fprintf(stderr, "NOT ENOUGH MEMORY...EXITING"); exit(-1); }
    tn->value = value;
    tn->parent = parent;
    tn->LC = NULL;
    tn->RC = NULL;
    return tn;
}

File: p2/src/TreeNode.java
public class TreeNode {

    private int value;
    private TreeNode parent;
    private TreeNode LC;
    private TreeNode RC;

	/**
	* Default TreeNode constructor
	*/
	public TreeNode(int value, TreeNode parent) {
		this.value  = value;
        this.parent = parent;
        this.LC     = null;
        this.RC     = null;
	}
}
